spinalcordtoolbox/testing/create_test_data.py

```diff
@@ -132,12 +133,29 @@ def dummy_segmentation(size_arr=(256, 256, 256), pixdim=(1, 1, 1), dtype=np.floa
     nx, ny, nz = [int(size_arr[i] * pixdim[i]) for i in range(3)]
     data = np.random.random((nx, ny, nz)) * 0.
     xx, yy = np.mgrid[:nx, :ny]
-    # loop across slices and add object
-    for iz in range(nz):
-        if shape == 'rectangle':  # theoretical CSA: (a*2+1)(b*2+1)
-            data[:, :, iz] = ((abs(xx - nx / 2) <= radius_RL) & (abs(yy - ny / 2) <= radius_AP)) * 1
-        if shape == 'ellipse':
-            data[:, :, iz] = (((xx - nx / 2) / radius_RL) ** 2 + ((yy - ny / 2) / radius_AP) ** 2 <= 1) * 1
+    if not interleaved:
+        # loop across slices and add object
+        for iz in range(nz):
+            if shape == 'rectangle':  # theoretical CSA: (a*2+1)(b*2+1)
+                data[:, :, iz] = ((abs(xx - nx / 2) <= radius_RL) & (abs(yy - ny / 2) <= radius_AP)) * 1
+            if shape == 'ellipse':
+                data[:, :, iz] = (((xx - nx / 2) / radius_RL) ** 2 + ((yy - ny / 2) / radius_AP) ** 2 <= 1) * 1
+    elif interleaved:
+        import numpy.matlib
+        from numpy import poly1d, polyfit
+        # define array based on a polynomial function, within Y-Z plane to simulate slicewise motion in A-P
+        y = np.matlib.repmat([round(nx / 2.) + pixdim[0]*factor, round(nx / 2.) - pixdim[0]*factor], 1, round(nz / 2))
+        if nz % 2 != 0:         # if z-dimension is odd, add one more element to fit size
+            y = numpy.append(y,round(nx / 2.) + pixdim[0]*factor)
+        y = y.reshape(nz)       # reshape to vector (1,R) -> (R,)
+        z = np.arange(0, nz)
+        p = poly1d(polyfit(z, y, deg=nz))
+        # loop across slices and add object
+        for iz in range(nz):
```

Comment:
code duplication with lines 138-142. you could create a nested function

Reply:
code slightly differs in second parentheses. Isn't current workaround better for clarity than nested function?